### PR TITLE
add win_initialize_disk module

### PR DESCRIPTION
**Pull request by branic**
##### SUMMARY

This module allows for initializing disks on a Windows system.

When needing to automate the creation of disks on a server to become usable after the disk is presented to the server it must be initialized before it can be partitioned and then formatted. There are existing modules to partition and format disks, this module adds the missing capability to initialize them.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

win_initialize_disk
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by ShachafGoldstein** [comment]
haven't gone over the whole code but would be great IMO to have path and Id options too and not only number as selectors

**Comment by branic** [comment]
@ShachafGoldstein That's a good idea. I'll add path and id as options.

File: lib/ansible/modules/windows/win_initialize_disk.ps1 [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$disk_number = $module.Params.disk_number
+$uniqueid = $module.Params.uniqueid
+$path = $module.Params.path
+$partition_style = $module.Params.style
+$bring_online = $module.Params.online
+$force_init = $module.Params.force
+
+function Get-AnsibleDisk {
```
Comment (ShachafGoldstein) [inline lib/ansible/modules/windows/win_initialize_disk.ps1:38]: I think you should allow for this function to get and return lists, i.e if i specify one disk by ID and another with a number - I want to Init both eventually

**Comment by ShachafGoldstein** [comment]
If you could also add check mode and idempotency, that would be amazing

**Comment by branic** [comment]
> If you could also add check mode and idempotency, that would be amazing

@ShachafGoldstein it already supports check mode and idempotency. Is there somewhere that you see where check mode isn't supported or is not idempotent? (or was this in reference to the request to support lists?)

**Comment by ShachafGoldstein** [comment]
Nvm my bad

File: lib/ansible/modules/windows/win_initialize_disk.ps1 (hunk 1 above)
Comment (branic) [reply lib/ansible/modules/windows/win_initialize_disk.ps1:38]: Allowing for lists would make the behavior of this module different from how the `win_format` and `win_partition` modules work (as well as the Unix equivalent modules such as `filesystem`).

The loop structure can be used on the task to provide a list with multiple disks and identifications. For example:

```yaml
  vars:
    - disks:
        - diskNumber: 1
          style: gpt
        - uniqueId: "60022480622E439002680C84CD374969"
        - path: '\\?\scsi#disk&ven_msft&prod_virtual_disk#2&1f4adffe&0&000001#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}'
          style: mbr

  tasks:
    - name: Initialize Disks
      win_initialize_disk:
        disk_number: '{{ item.diskNumber | default(omit) }}'
        uniqueid: '{{ item.uniqueId | default(omit) }}'
        path: '{{ item.path | default(omit) }}'
        style: '{{ item.style | default(omit) }}'
      loop: '{{ disks }}'

```

This also has the advantage that the other parameters can be specified differently for each disk.

Maybe this is a good candidate as an example in the documentation?

**Comment by branic** [comment]
@jborean93 @ShachafGoldstein any additional comments or thoughts on getting this merged?

**Comment by ShachafGoldstein** [comment]
None from me

**Comment by realged13** [comment]
glad to finally see this.  wish it made it to 2.9 :(

**Comment by branic** [comment]
@ShachafGoldstein @if-meaton @jborean93 @jhawkesworth @nitzmahone Can this get merged?

**Review by jborean93 — APPROVED** [review]
Really sorry for the delay, the changes here look good to me, thanks for working on this.

**Comment by branic** [comment]
Thanks @jborean93. I updated the documentation to fix the test failures.

**Comment by jborean93** [comment]
Thanks @branic for the quick fix.

**Pull request merged**